### ISSUE TITLE
Pipeline support in Cluster mode

### ISSUE THREAD

**Issue by doyshinda** [mitsuhiko/redis-rs#298]
Hi!

I'd like to take a stab at implementing support for pipelining when using the `ClusterConnection`. Playing around with the code a bit, my general idea of how to do this is:

1) instead of sending the entire pipeline as a packed command to the connection, send a vector of individual packed commands
2) group commands by the node they're supposed to talk to (maintaining an ordering of the original commands so they can be stitched back into a proper result vector)
3) build a "node command" that is equivalent of a packed pipeline command
4) for each node, send the node command (but don't read response(s) to take advantage of I/O concurrency)
5) for each node, read the node command response(s)
6) stitch the results from each node back into the original ordering
7) return result

Let me know me know if that roughly lines up with any ideas you had on how to implement this.

Cheers

EDIT: Removed the distinction between per-node commands and all nodes commands.

**Comment by doyshinda** [comment]
Implemented in #452 and #307

**Comment by sandeepyadavhas** [comment]
Hi @doyshinda , when I'm connecting redis-rs ClusterClient to a redis cluster, and check supports_pipelining, it is always returning false.

```
use std::error::Error;
use redis::cluster::ClusterClient;
use redis::{Commands, ConnectionLike};

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
	let nodes = vec![
		"redis://:host:port",
	];

	let client = ClusterClient::open(nodes).unwrap();
	let mut connection = client.get_connection().unwrap();
	if connection.supports_pipelining() {
		println!("connection supports pipelining")
	} else {
		println!("connection does not support pipelining")
	}

	Ok(())
}
```

After that I tried debugging the issue, by going in the implementation of redis-rs, and I found this line:

```
impl ConnectionLike for ClusterConnection {
    fn supports_pipelining(&self) -> bool {
        false
    }
```

Am I missing something, else can you help why even after #293 merge, I'm seeing static false returned for supports_pipelining function?

dependencies: redis = { version = "0.22.1", features = ["tokio-comp", "cluster"] }
